CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_CovertActionExpirationManager.uc

```diff
@@ -124,4 +151,18 @@ function AddActionExpirationInfo(StateObjectReference ActionRef, TDateTime Expir
 function RemoveActionExpirationInfo(ActionExpirationInfo ExpirationInfo)
 {
 	ActionExpirationInfoList.RemoveItem(ExpirationInfo);
+}
+
+function WarnOfExpiration(StateObjectReference WarningRef, XComGameState_CovertAction CovertAction)
+{
+	local int idx;
+	
+	for (idx = 0; idx < ActionExpirationInfoList.Length; idx++)
+	{
+		if (ActionExpirationInfoList[idx].ActionRef == WarningRef)
+		{
+			ActionExpirationInfoList[idx].bAlreadyWarnedOfExpiration = true;
```

Comment:
I like how you comment "can't have an open gamestate" but then edit a state variable here

Reply:
>.>